SystemManager/Src/SystemManager.cpp

```diff
@@ -0,0 +1,43 @@
+//
+// Created by Gagan Deep Singh on 2023-07-08.
+// Edited by Navtaj Hundal on 2023-07-23.
+//
+
+#include "SystemManager.hpp"
+#include "sbus_receiver.hpp"
+#include "sbus_defines.h"
+#include "rcreceiver_datatypes.h"
+#include "drivers_config.hpp"
+#include "independent_watchdog.h"
+#include "tim.h"
+
+SystemManager::SystemManager():
+    rcController_(&huart2),
+    throttleMotorChannel_(&htim2, TIM_CHANNEL_1),
+    yawMotorChannel_(&htim2, TIM_CHANNEL_2),
+    rollMotorChannel_(&htim2, TIM_CHANNEL_3),
+    pitchMotorChannel_(&htim2, TIM_CHANNEL_4),
+    watchdog_(&hiwdg)
+{}
+SystemManager::~SystemManager() {}
+
+void SystemManager::flyManually() {
+    for(;;){
+            updateRCInputs();
+            executeInputs();
+    }
+}
+
+void SystemManager::updateRCInputs() {
+    this->rcInputs_.assignValues(rcController_.GetRCControl());
+    watchdog_.refreshWatchdog();
+    return;
+}
+
+void SystemManager::executeInputs() {
+    this->throttleMotorChannel_.set(rcInputs_.throttle);
+    this->yawMotorChannel_.set(rcInputs_.yaw);
+    this->rollMotorChannel_.set(rcInputs_.roll);
+    this->pitchMotorChannel_.set(rcInputs_.pitch);
```

Comment:
2 notes
1. This assumes we only have 1 motor per axis, which wont be true for a plane. We will likely have 2 motors for roll.
2. There is no provision here for arm/disarm. The throttle motor will spin up if the throttle stick on the controller is bumped, and is not safe at all. (this is something that happens all the time, please add an arm/disarm check. Typically you can only arm if the throttle is down and a switch is hit.)

Reply:
1. I believe for M1 I was asked to assume that each axis has its own motor.  
2. Just added a provision to check for arm/disarm

Reply:
I can say with near certainty that you will need to add a second motor for the roll axis that is inverted (1 motor per aileron)